device/devicegate/filterHandler.go

```diff
@@ -0,0 +1,88 @@
+package devicegate
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/go-kit/kit/log/level"
+	"github.com/xmidt-org/webpa-common/logging"
+	"github.com/xmidt-org/webpa-common/xhttp"
+)
+
+// FilterHandler is an http.Handler that can get, add, and delete filters from a devicegate Interface
+type FilterHandler struct {
+	Gate Interface
+}
+
+func (fh *FilterHandler) ServeHTTP(response http.ResponseWriter, request *http.Request) {
+	logger := logging.GetLogger(request.Context())
+
+	method := request.Method
+	if method == http.MethodGet {
```

Comment:
In general, you don't want to check for things like HTTP method to perform custom logic.  Things like reading the body and parsing the URL form variables works properly no matter what HTTP method is used.  Leverage HTTP routing code, such as `http.ServeMux` or `gorilla/mux.Router`, to make decisions about what type of handling should be done.  That makes your HTTP handling code very generic and composable.

Here's two approaches to this.  Both are great, and it's a matter of personal taste which you use.

Approach 1: Different HTTP handlers
```
type MyHandler struct { /* a ServeHTTP method expecting a GET */ }
type FormHandler struct { /* a ServeHTTP method that looks at the content body */ }

r := mux.NewRouter()
r.Path("/foo").Methods("GET").Handler(new(MyHandler))
r.Path("/foo").Methods("POST").Handler(new(FormHandler))
```

Approach 2: Methods of the same type as handlers
```
type MyHandler struct {
    // stuff ....
}

func (mh *MyHandler) Simple(response http.ResponseWriter, request *http.Request) {
    // snip
}

func (mh *MyHandler) Form(response http.ResponseWriter, request *http.Request) {
    // snip
}

h := new(MyHandler)
r := mux.NewRouter()
r.Path("/foo").Methods("GET").HandlerFunc(h.Simple)
r.Path("/foo").Methods("POST").HandleFunc(h.Form)
```